web/packages/teleterm/src/ui/Search/useSearch.test.ts

```diff
@@ -63,4 +63,35 @@ describe('sortResults', () => {
     expect(fooMatches[0].score).toBeGreaterThan(quuxMatch.score);
     expect(fooMatches[1].score).toBeGreaterThan(quuxMatch.score);
   });
+
+  it('limits the results', () => {
+    const servers = Array(10)
+      .fill(undefined)
+      .map(() =>
+        makeResourceResult({
+          kind: 'server',
+          resource: makeServer({
+            labelsList: makeLabelsList({ foo: 'bar' }),
+          }),
+        })
+      );
+
+    // This item has a lower score
+    servers.unshift(
+      makeResourceResult({
+        kind: 'server',
+        resource: makeServer({
+          labelsList: makeLabelsList({ abc: 'bar123456' }),
+        }),
+      })
+    );
+
+    const sorted = sortAndLimitResults(servers, 'bar');
+
+    expect(sorted).toHaveLength(10);
+    // the item with the lowest score is not included
+    expect(
+      sorted.find(s => s.labelMatches.find(a => a.labelName === 'abc'))
```

Comment:
What's the advantage of doing it this way vs giving the lowest result a unique URI and then checking if a result with that URI is not present in the final array?

Reply:
You're right, it can be done more simply.